fscbco/topnews
# Internal team news feed with stars ⭐️

### Intro
Hey, my name is Alex, and here is my submission for the code challenge! I left some notes as comments on the PR 👋

### Preview
**Sign In**
![Screen Shot 2022-12-11 at 11 27 59 AM](https://user-images.githubusercontent.com/15987080/206916007-394dd75c-7a25-4694-a937-22b580d7277b.png)

**All**
![Screen Shot 2022-12-11 at 11 28 31 AM](https://user-images.githubusercontent.com/15987080/206916008-2c8d24cb-388c-47ca-a0e5-085021d8ba86.png)

**Starred**
![Screen Shot 2022-12-11 at 11 28 48 AM](https://user-images.githubusercontent.com/15987080/206916010-dfee258e-0f22-474e-a7e0-735c61e7aeb5.png)

**Search**
![Screen Shot 2022-12-11 at 11 29 00 AM](https://user-images.githubusercontent.com/15987080/206916005-7791caf0-99b4-4436-9150-7a1d5954096a.png)

### Description
This PR adds a new "feature" that pulls in posts from [Hacker News](https://news.ycombinator.com/) and allows internal team members to star them. Starring is similar to bookmarking - it marks a post as interesting. You can search posts by title and view only starred posts.

### Steps to test
1. Pull down this branch locally.
2. Install new gems with `bundle install`.
3. Re-seed your local database - `rails db:seed`.
4. Fire up the server locally with `bin/dev`.
5. Visit [http://localhost:3000](http://localhost:3000) to check out the app 🎉 

![shooting_finger_stars_gif](https://media.giphy.com/media/pULj9CGmIcfuzXTPfp/giphy.gif)
